src/Common.sol

```diff
@@ -691,4 +691,19 @@ abstract contract Common is AccessControl, Pausable {
     function getMaxFeeX64(FeeType feeType) public view returns (uint64) {
         return _maxFeeX64[feeType];
     }
+
+    /// @dev some tokens require allowance == 0 to approve new amount
+    /// but some tokens does not aloow approve ammount = 0
+    /// we try to set allowence = 0 before approve new amount. if it revert means that
+    /// the token not allow to approve 0, which means the following line code will work properly
+    function _safeApprove(IERC20 token, address _spender, uint256 _value) internal {
+        bytes memory approveData = abi.encodeWithSelector(token.approve.selector, spender, 0);
+        try address(token).functionCall(approveData, "SafeERC20: low-level call approve failed") {} catch {}
+
+        approveData = abi.encodeWithSelector(token.approve.selector, _spender, _value);
```

Comment:
if `_value` is 0, do we need to approve twice?

Reply:
fixed